dev/release/verify-release-candidate.sh

```diff
@@ -468,6 +468,22 @@ install_conda() {
   conda activate base
 }
 
+maybe_setup_maven() {
+  MAVEN_VERSION=3.8.7
+  SYSTEM_MAVEN_VERSION=$(mvn -v | head -n 1 | awk '{print $3}')
+  if [[ $(echo "$SYSTEM_MAVEN_VERSION $MAVEN_VERSION" | awk '{print ($1 < $2)}') -eq 1 ]]; then
```

Comment:
Problem here: If any of the version parts have 2 digits (which [can happen](https://maven.apache.org/docs/history.html)), this could fail. For example:
```sh
echo "3.8.10 3.8.7" | awk '{print ($1 < $2)}'
```
gives `1`, but we wouldn't want to upgrade in that case.

Reply:
Looking into it.

Reply:
@ianmcook looking into this I found https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN8855

If we go for `alpha`, `beta` like versioning, the logic gets a bit complicated and it is not as straightforward as it is done for node since it is numbers. I wrote a few versions of version checkers now and tested. I am still verifying...

Reply:
How about using `sort --version-sort`?

```bash
  older_version=$((echo $SYSTEM_MAVEN_VERSION; echo $MAVEN_VERSION) | sort --version-sort | head -n1) 
  if [[ "$older_version" = "$SYSTEM_MAVEN_VERSION" ]]; then
```

Reply:
I tried that, but can it determine `gamma` vs `delta`?

Reply:
Oh wait...

Reply:
Why do we need to care about `gamma`, `delta` and so on?
I think that we don't use them for `MAVEN_VERSION`.

Reply:
Would it work for the following

```bash
# Desired Maven version
MAVEN_VERSION="3.0.0-gamma-13"

# System's Maven version
SYSTEM_MAVEN_VERSION="3.0.0-delta-13"
```

Reply:
`3.0.0-gamma-13` is an unreleased Maven version, right?
If so, we should not use the version in `MAVEN_VERSION`. We should use only released Maven.

Reply:
`3.0.0-gamma-13` merely using as an example to test a point. 

> If so, we should not use the version in MAVEN_VERSION. We should use only released Maven.

Yes, we can agree to this and only verify released versions.

Reply:
@kou CIs are passing. I added the version check suggestion. Could you please check?

Reply:
Looks good to me. I added a fix in https://github.com/apache/arrow/pull/40514/commits/d7cae29b7e50404f550e3ac48aa2640e2177776a to handle the case of unreleased versions.